lib/migrations/createGenesisDao.ts

```diff
@@ -23,135 +29,130 @@ interface FounderSpec {
    */
   reputation: string | number;
 }
+
+export interface ForgedDaoInfo {
+  avatarAddress: Address;
+  orgName: string;
+  daoCreatorInst: any;
+}
+
 /**
  * Migration callback
  */
-export const arcJsDeployer = (web3: Web3, artifacts: any, deployer: any): void => {
-
-  // so Utils.getWeb3 can find it
-  (global as any).web3 = web3;
-
-  const network = env.arcjs_network || "ganache";
-
-  const internalFoundersConfigLocation = "../../migrations/founders.json";
-  const foundersConfig = require(internalFoundersConfigLocation).founders;
+export class GenesisDaoCreator {
 
-  const customFoundersConfigLocation = env.arcjs_foundersConfigurationLocation || internalFoundersConfigLocation;
-
-  if (internalFoundersConfigLocation !== customFoundersConfigLocation) {
-    console.log(`merging custom founders from ${customFoundersConfigLocation}`);
-    const customFoundersConfig = require(customFoundersConfigLocation).founders;
-    // merge the two
-    Object.assign(foundersConfig, customFoundersConfig);
-  }
-
-  const founders = foundersConfig[network];
-
-  if (!founders || (founders.length === 0)) {
-    throw new Error(`no founders were given for the network: ${network}`);
+  constructor(
+    private web3: Web3,
+    private network: string) {
   }
 
   /**
-   * Truffle Solidity artifact wrappers
+   * Forge the Genesis DAO.  Note this does not set the schemes.
    */
-  const AbsoluteVote = artifacts.require("AbsoluteVote.sol");
-  const Avatar = artifacts.require("Avatar.sol");
-  const ContributionReward = artifacts.require("ContributionReward.sol");
-  const DaoCreator = artifacts.require("DaoCreator.sol");
-  // ExecutableTest is used only by tests
-  const ExecutableTest = artifacts.require("ExecutableTest.sol");
-  const GlobalConstraintRegistrar = artifacts.require("GlobalConstraintRegistrar.sol");
-  const SchemeRegistrar = artifacts.require("SchemeRegistrar.sol");
-  const SimpleICO = artifacts.require("SimpleICO.sol");
-  const TokenCapGC = artifacts.require("TokenCapGC.sol");
-  const UController = artifacts.require("UController.sol");
-  const UpgradeScheme = artifacts.require("UpgradeScheme.sol");
-  const VestingScheme = artifacts.require("VestingScheme.sol");
-  const VoteInOrganizationScheme = artifacts.require("VoteInOrganizationScheme.sol");
-  const GenesisProtocol = artifacts.require("GenesisProtocol.sol");
-  const ControllerCreator = artifacts.require("ControllerCreator.sol");
+  public async forge(foundersConfigurationLocation: string): Promise<ForgedDaoInfo> {
 
-  /**
-   * Pattern for using async/await found here:
-   *  https://github.com/trufflesuite/truffle/issues/501#issuecomment-332589663
-   */
-  deployer.then(async () => {
+    const live = this.network === "live";
     /**
-     *  Genesis DAO parameters,  FOR TESTING PURPOSES ONLY
+     * Genesis DAO parameters
      */
-    const orgName = "Genesis Alpha";
-    const tokenName = "Genesis Alpha";
+    const orgName = live ? "Genesis" : "Genesis Alpha";
```

Comment:
Why "Genesis" ? Why "GenesisAlpha"  Alpha ?
How about rename this script to createDao and pass to it a params (
e.g orgName... ).
Then this script can be used to deploy any Dao .

Reply:
Seems like a reasonable feature request to me, but I would prefer to make that a separate PR,  Is Ok with you?

Reply:
why separate PR ?
This should be in this pr.

Reply:
Why "GenesisAlpha" Alpha ?
It is just another DAO